app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt

```diff
@@ -118,6 +118,31 @@ class ThreadListAdapter @Inject constructor(
         this.recyclerView = recyclerView
     }
 
+    override fun getItemViewType(position: Int): Int = runCatchingRealm {
+        val item = dataSet[position]
+        return when {
+            item is String -> DisplayType.DATE_SEPARATOR.layout
+            item is FolderRole -> DisplayType.FLUSH_FOLDER_BUTTON.layout
+            item is Unit -> DisplayType.LOAD_MORE_BUTTON.layout
+            displaySeeAllButton -> DisplayType.SEE_ALL_BUTTON.layout
+            else -> DisplayType.THREAD.layout
+        }
+    }.getOrDefault(super.getItemViewType(position))
+
+    override fun getItemId(position: Int): Long = runCatchingRealm {
+        return when (val item = dataSet[position]) {
+            is Thread -> item.uid.hashCode().toLong()
+            is String -> item.hashCode().toLong()
+            else -> super.getItemId(position)
+        }
+    }.getOrDefault(super.getItemId(position))
+
+    fun getItemPosition(threadUid: String): Int? {
+        return dataSet
+            .indexOfFirst { it is Thread && it.uid == threadUid }
+            .takeIf { position -> position != -1 }
+    }
+
```

Comment:
Maybe we can group `override` methods together?

Reply:
We sort them by logic.